src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList';
+import { TodoFilter, StatusOfFilter } from '../TodoFilter';
+import { useLocalStorage } from '../../helpers/useLocalStorage';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [value, setValue] = useState('');
+  const [filter, setFilter] = useState(StatusOfFilter.All);
+
+  const notCompletedTogglers = todos.filter(
+    todo => !todo.completed,
+  );
+
+  const completedTogglers = todos.filter(
+    todo => todo.completed,
+  );
+
+  const updatedTodos = (newTodo: Todo) => {
+    setTodos([...todos, newTodo]);
+  };
+
+  const clearCompletedTodos = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    switch (filter) {
+      case StatusOfFilter.All:
+        return true;
+      case StatusOfFilter.Active:
+        return !todo.completed;
+      case StatusOfFilter.Completed:
+        return todo.completed;
+      default:
+        throw new Error('Unexpected value');
+    }
+  });
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+
+      if (event.target.value) {
+        updatedTodos({
+          id: +new Date(),
+          title: event.target.value,
+          completed: false,
+        });
+      }
+
+      setValue('');
+    }
+  };
+
```

Comment:
The parameter name `event` in `handleToggleAll` is misleading, as it's a boolean value. Consider renaming it to `checked` for clarity. ✏️